raftpb/raft.proto

```diff
@@ -71,22 +75,32 @@ message Message {
 	optional uint64      from        = 3  [(gogoproto.nullable) = false];
 	optional uint64      term        = 4  [(gogoproto.nullable) = false];
 	// logTerm is generally used for appending Raft logs to followers. For example,
-	// (type=MsgApp,index=100,logTerm=5) means leader appends entries starting at
-	// index=101, and the term of entry at index 100 is 5.
+	// (type=MsgApp,index=100,logTerm=5) means the leader appends entries starting
+	// at index=101, and the term of the entry at index 100 is 5.
 	// (type=MsgAppResp,reject=true,index=100,logTerm=5) means follower rejects some
 	// entries from its leader as it already has an entry with term 5 at index 100.
+	// (type=MsgStorageAppendResp,index=100,logTerm=5) means the local node wrote
+	// entries up to index=100 in stable storage, and the term of the entry at index
+	// 100 was 5.
 	optional uint64      logTerm     = 5  [(gogoproto.nullable) = false];
 	optional uint64      index       = 6  [(gogoproto.nullable) = false];
 	repeated Entry       entries     = 7  [(gogoproto.nullable) = false];
 	optional uint64      commit      = 8  [(gogoproto.nullable) = false];
+	// hardState can be non-nil for MsgStorageAppend messages and is nil for all
+	// other message types.
+	optional HardState   hardState   = 13 [(gogoproto.nullable) = true];
```

Comment:
Where did we land on https://github.com/etcd-io/etcd/pull/14627#discussion_r1015477471?

Reply:
I had added a TODO to `newStorageAppendMsg` to revisit this suggestion. Let me do that now. I agree that we'll want to resolve it before merging because this is part of the external interface of the library.

Reply:
Done. I think I like it, but let me know what you think about the new commit at the end (`store Vote in MsgStorageAppend, not *HardState`).